### ISSUE TITLE
serverless offline does not work with aws: and ssm: variables

### ISSUE THREAD

**Issue by lqueryvg** [serverless/serverless#11674]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

It's impossible to work with serverless offline if your `serverless.yml` file contains any variables which resolve from AWS (e.g. `${aws:accountId}` or `${ssm:/some/path}`).

You get errors like the following:

```
  - Cannot resolve variable at "custom.key1": AWS provider credentials not found. Learn how to set up AWS provider credentials in our docs here: <http://slss.io/aws-creds-setup>.,
```

The only workaround I've seen is to put all such variables in a separate file - and have a separate file for every environment.  This is cludgey, nasty, horrible ....

I don't know whether this is a bug in serverless or serverless-offline, or potentially a bit of both.

It should work like this:

1. serverless-offline should have the common sense to not try to connect to an AWS service when running in offline mode - you shouldn't need to set `AWS_PROFILE` and configure dummy credentials in your `.aws` directory
2. by default it should replace these variables with a default value, e.g. `123412341234` for the `aws:accountId`
3. optionally it should be possible to override those default values via keys in the custom section of the serverless.yml

This is how plugins like `serverless-offline-ssm` and `serverless-pseudo-parameters` work.
But the recommendation seems to be to use the native serverless features instead, despite the fact that they just don't work offline !

### Service configuration (serverless.yml) content

```yaml
service: testcase

plugins:
  - serverless-offline

provider:
  name: aws

custom:
  key1: ${aws:accountId}
  key2: ${ssm:/some/path}
```

### Command name and used flags

yarn exec -- sls offline start

### Command output

```shell
❯ yarn exec -- sls offline start
yarn exec v1.22.19
Environment: darwin, node 16.17.0, framework 3.23.0 (local), plugin 6.2.2, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
Cannot resolve serverless.yml: Variables resolution errored with:
  - Cannot resolve variable at "custom.key1": AWS provider credentials not found. Learn how to set up AWS provider credentials in our docs here: <http://slss.io/aws-creds-setup>.,
  - Cannot resolve variable at "custom.key2": AWS provider credentials not found. Learn how to set up AWS provider credentials in our docs here: <http://slss.io/aws-creds-setup>.
error Command failed.
Exit code: 1
Command: sls
Arguments: offline start
Directory: <REDACTED>
Output:

info Visit https://yarnpkg.com/en/docs/cli/exec for documentation about this command.
```

### Environment information

```shell
Framework Core: 3.23.0 (local)
Plugin: 6.2.2
SDK: 4.3.2
```

**Comment by medikoo** [comment]
@lqueryvg please report at `serverless-offline` repository

**Comment by lqueryvg** [comment]
OK. I will.
But are you sure this is serverless-offline and not serverless ?

**Comment by medikoo** [comment]
Yes, Framework works without issues. You have issue with functionality provided by the plugin

**Comment by lqueryvg** [comment]
But this issue was already closed but the serverless-offline people without any resolution:
https://github.com/dherault/serverless-offline/issues/1278

That issue has a comment which suggests it's a problem with the framework, not the plugin:
https://github.com/dherault/serverless-offline/issues/1278#issuecomment-1181203314

**Comment by medikoo** [comment]
It's offline plugin that hacks the process so things work offline. Same way it should approach variables resolution.

If plugin implementors see some hacking difficult to do. They can open the issue saying: _We want to override the resolution of `aws:` variables, yet we're not sure how to tackle that. What's the recommendation from the Framework team_
